common/src/main/scala/org/neo4j/spark/converter/DataConverter.scala

```diff
@@ -77,29 +77,23 @@ object SparkToNeo4jDataConverter {
 class SparkToNeo4jDataConverter extends DataConverter[Value] {
 
   override def convert(value: Any, dataType: DataType): Value = {
-    dataType match {
-      case _: DayTimeIntervalType if value != null =>
-        return SparkToNeo4jDataConverter.dayTimeIntervalToNeo4j(value.asInstanceOf[Long])
-      case _: YearMonthIntervalType if value != null =>
-        return SparkToNeo4jDataConverter.yearMonthIntervalToNeo4j(value.asInstanceOf[Int])
-      case _ => // do nothing
-    }
-
     value match {
       case date: java.sql.Date           => convert(date.toLocalDate, dataType)
-      case timestamp: java.sql.Timestamp => convert(timestamp.toLocalDateTime, dataType)
+      case timestamp: java.sql.Timestamp => convert(timestamp.toInstant.atZone(ZoneOffset.UTC), dataType)
       case intValue: Int if dataType == DataTypes.DateType =>
         convert(
           DateTimeUtils
             .toJavaDate(intValue),
           dataType
         )
+      case intValue: Int if dataType.isInstanceOf[YearMonthIntervalType] =>
+        SparkToNeo4jDataConverter.yearMonthIntervalToNeo4j(intValue)
       case longValue: Long if dataType == DataTypes.TimestampType =>
-        convert(
-          DateTimeUtils
-            .toJavaTimestamp(longValue),
-          dataType
-        )
+        convert(DateTimeUtils.toJavaTimestamp(longValue), dataType)
+      case longValue: Long if dataType == DataTypes.TimestampNTZType =>
+        convert(DateTimeUtils.microsToLocalDateTime(longValue), dataType)
```

Comment:
question: why micros? the docs only mention second precision

Reply:
I reckon the spark-side is microseconds:

https://api-docs.databricks.com/scala/spark/latest/org/apache/spark/sql/types/TimestampNTZType.html

Reply:
ah, https://spark.apache.org/docs/latest/sql-ref-datatypes.html is not up-to-date, then.